### PR TITLE
Add pyyaml dependency and add install tests workflow

### PR DESCRIPTION
**Pull request by gsheni**
- Add a install test workflow which install core sdv requirements, imports sdv, and gets the version
- Add pyyaml dependency

### CODE REVIEW

File: .github/workflows/install.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  push:
+    branches:
+      - main
+jobs:
+  install:
+    name: ${{ matrix.os }} - ${{ matrix.python_version }} install
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-latest, macos-latest]
```
Comment (amontanez24) [inline .github/workflows/install.yaml:14]: let's just run on ubuntu

Comment (frances-h) [inline .github/workflows/install.yaml:14]: should we add windows as well?

**Review by amontanez24 — APPROVED** [review]

File: .github/workflows/install.yaml (hunk 1 above)
Comment (gsheni) [reply .github/workflows/install.yaml]: Fixed:
https://github.com/sdv-dev/SDV/pull/2007/commits/e7b01d98d0302fdb0c6eb7f4953deddc6242769f

Comment (gsheni) [reply .github/workflows/install.yaml]: Doing just ubuntu for now to keep testing to a minimal

**Review by frances-h — APPROVED** [review]

**Pull request merged**